### DISCUSSION TITLE
Question about link static lib.

### DISCUSSION THREAD

**Discussion by sonygod** [emscripten-core/emscripten#18194] [Q&A]
Hello, how to link static lib in emcc?

example:

     gcc test.c -lgc   //work

     cc  test.c -lgc //work

 
and emcc

    emcc test.c -lgc //error.

how to link libgc?

and sysinfo

  ubuntu 22.04

  emcc 3.1.25

**Comment by sbc100** [comment]
Static libraries should work fine under emscripten.  What is the error you are seeing?  What is libgc.a? 

Bear in mind that emscripten cannot use object files or libraries from your system.. any libraries you want to use with emscripten would  need to be compiled with the emcc compiler.

**Comment by sonygod** [comment]
oh,I see, you mean libgc-dev has to build by emcc .


~~@sbc100  hello , [ libgc-dev](https://zoomadmin.com/HowToInstall/UbuntuPackage/libgc-dev)~~

**Comment by sbc100** [comment]
BTW you might stuggle to get boehm GC working under wasm/emscripten, because scanning the stack is much harder.  The actual value and execution stack is not visible to the application, so you would need to deliberately spill all your stack pointers to linear memory.   See https://github.com/WebAssembly/binaryen/issues/4566
